R/error-checks.R

```diff
@@ -0,0 +1,162 @@
+isSameLength <- function(...) {
+  args <- list(...)
+  nrOfLengths <- length(unique(lengths(args)))
+
+  return(nrOfLengths == 1)
+}
+
+#' Check if the provided object has nbElements elements
+#'
+#' @param object An object or a list of objects
+#' @param nbElements number of elements that are supposed in object
+#'
+#' @return TRUE if the object or all objects inside the list have nbElements.
+#' Only the first level of the given list is considered.
+isOfLength <- function(object, nbElements) {
+  return(length(object) == nbElements)
+}
+
+validateIsOfLength <- function(object, nbElements) {
+  if (isOfLength(object, nbElements)) {
+    return()
+  }
+  stop(messages$errorWrongLength(object, nbElements))
+}
+
+#' Check if the provided object is of certain type
+#'
+#' @param object An object or a list of objects
+#' @param type String  representation or Class of the type that should be checked for
+#'
+#' @return TRUE if the object or all objects inside the list are of the given type.
+#' Only the first level of the given list is considered.
+isOfType <- function(object, type) {

```

Comment:
Let's copy all the tests from Core for now. Better to have it duplicate dif we duplicate the code as well